### PR TITLE
Don't ignore return value of functions

### PR DESCRIPTION
**Pull request by hossam-nasr**
This PR makes it such that return values of functions are not ignored, and instead set as the `returnValue` property of the gRPC `functionInvocationResponse` message, even if no explicit output binding is set. This helps in cases such as Durable Functions where the function invocation is invoked programmatically and the return value is used without an explicit output binding

### CODE REVIEW

File: src/InvocationModel.ts [hunk 1]
```diff
@@ -104,6 +104,10 @@ export class InvocationModel implements coreTypes.InvocationModel {
             }
         }
 
+        if (!response.returnValue && response.outputData.length == 0) {
```
Comment (ejizba) [inline src/InvocationModel.ts]: I tested this on an http trigger with no output binding and it _is_ returning the value. It literally just shows the json as the response which seems pretty weird
Screen Shot 2022-10-07 at 2 14 08 PM

In other words, I do think we have to be more careful on this. Possible solutions include
1. Only set return value here based on trigger (either "not http" or "is durable"). This would match old behavior
2. Ask durable to pass in a flag somewhere saying "returnIfNoBindingSet". This would require changes in both repos
3. Ask durable to configure an output binding under the covers for the user. This would only require changes in durable

**Review by ejizba — CHANGES_REQUESTED** [review]

File: src/InvocationModel.ts (hunk 1 above)
Comment (hossam-nasr) [reply src/InvocationModel.ts]: I would favor the first option (specifically the "if not HTTP" part) since it matches old behavior anyway. Option 3 isn't really feasible because there isn't a valid output binding type that Durable can set

Comment (ejizba) [reply src/InvocationModel.ts]: I'm okay with the first option to unblock durable, although I would still prefer if we had a better reason than "it matches old behavior". The old behavior had a lot of weird things that did not make sense

Comment (hossam-nasr) [reply src/InvocationModel.ts]: Went with the first option to unblock Durable and added a detailed comment on why this is useful

File: .eslintrc.json [hunk 2]
```diff
@@ -1,10 +1,6 @@
 {
```
Comment (ejizba) [inline .eslintrc.json]: Please revert these unrelated changes, otherwise looks good

**Review by ejizba — APPROVED** [review]
